### ISSUE TITLE
Getting "Undetermined" as Caption option for iOS

### ISSUE THREAD

**Issue by marcosthejew** [shaka-project/shaka-player#5782]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**If the problem is related to FairPlay, have you read the tutorial?**

Yes

**What version of Shaka Player are you using?**

4.4.3

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `main`?**
Yes

**Are you using the demo app or your own custom app?**
Custom app

**If custom app, can you reproduce the issue using our demo app?**
No

**What browser and OS are you using?**
Safari and Chrome iOS 17.0.3

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
N/A

**What are the manifest and license server URIs?**

https://demo.unified-streaming.com/k8s/features/stable/video/tears-of-steel/tears-of-steel.ism/.m3u8

**What configuration are you using?  What is the output of `player.getConfiguration()`?**

`'{\n  "drm": {\n    "retryParameters": {\n      "maxAttempts": 2,\n      "baseDelay": 1000,\n      "backoffFactor": 2,\n      "fuzzFactor": 0.5,\n      "timeout": 30000,\n      "stallTimeout": 5000,\n      "connectionTimeout": 10000\n    },\n    "servers": {},\n    "clearKeys": {},\n    "advanced": {},\n    "delayLicenseRequestUntilPlayed": false,\n    "persistentSessionOnlinePlayback": false,\n    "persistentSessionsMetadata": [],\n    "logLicenseExchange": false,\n    "updateExpirationTime": 1,\n    "preferredKeySystems": [],\n    "keySystemsMapping": {},\n    "parseInbandPsshEnabled": false,\n    "minHdcpVersion": ""\n  },\n  "manifest": {\n    "retryParameters": {\n      "maxAttempts": 2,\n      "baseDelay": 1000,\n      "backoffFactor": 2,\n      "fuzzFactor": 0.5,\n      "timeout": 30000,\n      "stallTimeout": 5000,\n      "connectionTimeout": 10000\n    },\n    "availabilityWindowOverride": null,\n    "disableAudio": false,\n    "disableVideo": false,\n    "disableText": false,\n    "disableThumbnails": false,\n    "defaultPresentationDelay": 0,\n    "segmentRelativeVttTiming": false,\n    "raiseFatalErrorOnManifestUpdateRequestFailure": false,\n    "dash": {\n      "clockSyncUri": "",\n      "ignoreDrmInfo": false,\n      "disableXlinkProcessing": false,\n      "xlinkFailGracefully": false,\n      "ignoreMinBufferTime": false,\n      "autoCorrectDrift": true,\n      "initialSegmentLimit": 1000,\n      "ignoreSuggestedPresentationDelay": false,\n      "ignoreEmptyAdaptationSet": false,\n      "ignoreMaxSegmentDuration": false,\n      "keySystemsByURI": {\n        "urn:uuid:1077efec-c0b2-4d02-ace3-3c1e52e2fb4b": "org.w3.clearkey",\n        "urn:uuid:e2719d58-a985-b3c9-781a-b030af78d30e": "org.w3.clearkey",\n        "urn:uuid:edef8ba9-79d6-4ace-a3c8-27dcd51d21ed": "com.widevine.alpha",\n        "urn:uuid:9a04f079-9840-4286-ab92-e65be0885f95": "com.microsoft.playready",\n        "urn:uuid:79f0049a-4098-8642-ab92-e65be0885f95": "com.microsoft.playready",\n        "urn:uuid:f239e769-efa3-4850-9c16-a903c6932efb": "com.adobe.primetime"\n      },\n      "sequenceMode": false\n    },\n    "hls": {\n      "ignoreTextStreamFailures": false,\n      "ignoreImageStreamFailures": false,\n      "defaultAudioCodec": "mp4a.40.2",\n      "defaultVideoCodec": "avc1.42E01E",\n      "ignoreManifestProgramDateTime": false,\n      "mediaPlaylistFullMimeType": "video/mp2t; codecs=\\"avc1.42E01E, mp4a.40.2\\"",\n      "useSafariBehaviorForLive": true,\n      "liveSegmentsDelay": 3,\n      "sequenceMode": true,\n      "ignoreManifestTimestampsInSegmentsMode": false\n    },\n    "mss": {\n      "sequenceMode": false,\n      "keySystemsBySystemId": {\n        "9a04f079-9840-4286-ab92-e65be0885f95": "com.microsoft.playready",\n        "79f0049a-4098-8642-ab92-e65be0885f95": "com.microsoft.playready"\n      }\n    }\n  },\n  "streaming": {\n    "retryParameters": {\n      "maxAttempts": 2,\n      "baseDelay": 1000,\n      "backoffFactor": 2,\n      "fuzzFactor": 0.5,\n      "timeout": 30000,\n      "stallTimeout": 5000,\n      "connectionTimeout": 10000\n    },\n    "rebufferingGoal": 2,\n    "bufferingGoal": 10,\n    "bufferBehind": 30,\n    "ignoreTextStreamFailures": false,\n    "alwaysStreamText": false,\n    "startAtSegmentBoundary": false,\n    "gapDetectionThreshold": 0.5,\n    "gapJumpTimerTime": 0.25,\n    "durationBackoff": 1,\n    "safeSeekOffset": 5,\n    "stallEnabled": true,\n    "stallThreshold": 1,\n    "stallSkip": 0.1,\n    "useNativeHlsOnSafari": true,\n    "inaccurateManifestTolerance": 2,\n    "lowLatencyMode": false,\n    "autoLowLatencyMode": false,\n    "forceHTTPS": false,\n    "preferNativeHls": false,\n    "updateIntervalSeconds": 1,\n    "dispatchAllEmsgBoxes": false,\n    "observeQualityChanges": false,\n    "maxDisabledTime": 30,\n    "parsePrftBox": false,\n    "segmentPrefetchLimit": 0,\n    "liveSync": false,\n    "liveSyncMaxLatency": 1,\n    "liveSyncPlaybackRate": 1.1\n  },\n  "mediaSource": {\n    "sourceBufferExtraFeatures": "",\n    "forceTransmux": false,\n    "insertFakeEncryptionInInit": true\n  },\n  "offline": {\n    "usePersistentLicense": true,\n    "numberOfParallelDownloads": 5\n  },\n  "abr": {\n    "enabled": true,\n    "useNetworkInformation": true,\n    "defaultBandwidthEstimate": 1000000,\n    "switchInterval": 8,\n    "bandwidthUpgradeTarget": 0.85,\n    "bandwidthDowngradeTarget": 0.95,\n    "restrictions": {\n      "minWidth": 0,\n      "maxWidth": null,\n      "minHeight": 0,\n      "maxHeight": null,\n      "minPixels": 0,\n      "maxPixels": null,\n      "minFrameRate": 0,\n      "maxFrameRate": null,\n      "minBandwidth": 0,\n      "maxBandwidth": null\n    },\n    "advanced": {\n      "minTotalBytes": 128000,\n      "minBytes": 16000,\n      "fastHalfLife": 2,\n      "slowHalfLife": 5\n    },\n    "restrictToElementSize": false,\n    "restrictToScreenSize": false,\n    "ignoreDevicePixelRatio": false,\n    "clearBufferSwitch": false,\n    "safeMarginSwitch": 0\n  },\n  "autoShowText": 3,\n  "preferredAudioLanguage": "",\n  "preferredAudioLabel": "",\n  "preferredTextLanguage": "",\n  "preferredVariantRole": "",\n  "preferredTextRole": "",\n  "preferredAudioChannelCount": 2,\n  "preferredVideoHdrLevel": "AUTO",\n  "preferredVideoCodecs": [],\n  "preferredAudioCodecs": [],\n  "preferForcedSubs": false,\n  "preferredDecodingAttributes": [],\n  "restrictions": {\n    "minWidth": 0,\n    "maxWidth": null,\n    "minHeight": 0,\n    "maxHeight": null,\n    "minPixels": 0,\n    "maxPixels": null,\n    "minFrameRate": 0,\n    "maxFrameRate": null,\n    "minBandwidth": 0,\n    "maxBandwidth": null\n  },\n  "playRangeStart": 0,\n  "playRangeEnd": null,\n  "cmcd": {\n    "enabled": false,\n    "sessionId": "",\n    "contentId": "",\n    "useHeaders": false\n  },\n  "lcevc": {\n    "enabled": false,\n    "dynamicPerformanceScaling": true,\n    "logLevel": 0,\n    "drawLogo": false\n  },\n  "ads": {\n    "customPlayheadTracker": false\n  }\n}'`

**What did you do?**

Create the HTML file: 
```
<!DOCTYPE html>
<html>
  <head>
    <!-- Shaka Player compiled library: -->
    <script src=https://cdnjs.cloudflare.com/ajax/libs/mux.js/7.0.0/mux.min.js></script>
    <script src="https://ajax.googleapis.com/ajax/libs/shaka-player/4.4.2/shaka-player.ui.js"></script>
    <!-- Your application source: -->
    <script src="myapp.js"></script>
    <link rel="stylesheet" type="text/css" href="dist/controls.css">
  </head>
  <body>
    <div id="videoContainer" style="width:640px; height:360px;">
    <video id="video"
           poster="https://shaka-player-demo.appspot.com/assets/poster.jpg"
           controls autoplay></video>
          </div>
  </body>
</html>
```
Create the file: `myapp.js` with the contents:

```
const manifestUri =
    'https://demo.unified-streaming.com/k8s/features/stable/video/tears-of-steel/tears-of-steel.ism/.m3u8';

function initApp() {
  // Install built-in polyfills to patch browser incompatibilities.
  shaka.polyfill.installAll();

  // Check to see if the browser supports the basic APIs Shaka needs.
  if (shaka.Player.isBrowserSupported()) {
    // Everything looks good!
    initPlayer();
  } else {
    // This browser does not have the minimum set of APIs we need.
    console.error('Browser not supported!');
  }
}

async function initPlayer() {
  // Create a Player instance.
  const video = document.getElementById('video');
  const player = new shaka.Player(video);

  // Attach player to the window to make it easy to access in the JS console.
  window.player = player;
  //debugger;
  // Listen for error events.
  player.addEventListener('error', onErrorEvent);


  // Try to load a manifest.
  // This is an asynchronous process.
  try {
    await player.load(manifestUri);
    // This runs if the asynchronous load is successful.
    console.log('The video has now been loaded!');
    shaka.polyfill.installAll();
    shaka.polyfill.PatchedMediaKeysApple.install();
    const videoContainer = document.getElementById('videoContainer');
    const ui = new shaka.ui.Overlay(player, videoContainer, player.getMediaElement());
    const config = {
            controlPanelElements: [
                'play_pause',
                'spacer',
                'time_and_duration',
                'volume',
                'mute',
                'picture_in_picture',
                'fullscreen',
                'captions',
            ],
        };
        ui.configure(config);
        ui.setEnabled(true);

    player.addTextTrackAsync('https://raw.githubusercontent.com/1c7/vtt-test-file/master/vtt%20files/2.%20No%20Index%20Number%2C%20download%20using%20youtube-dl.vtt', 'en', 'captions', 'text/vtt').then(function(track) {
    }).catch(function error(error){
      console.log(error);
  });
  
  } catch (e) {
    // onError is executed if the asynchronous load fails.
    onError(e);
  }
}

function onErrorEvent(event) {
  // Extract the shaka.util.Error object from the event.
  onError(event.detail);
}

function onError(error) {
  // Log the error.
  console.error('Error code', error.code, 'object', error);
}

document.addEventListener('DOMContentLoaded', initApp);

```

**What did you expect to happen?**
Only one caption track getting added with the "English" label.

**What actually happened?**
I get a (default) caption option titled "Undetermined". Only on Windowed mode.
 
![image](https://github.com/shaka-project/shaka-player/assets/1500967/6cb46a7a-7e3e-41c4-bdf6-477f77e9df5b)

**Comment by avelad** [comment]
It seems that Apple analyzes the video segments and detects some available CC and automatically put the "bad" track. You can see more information here: https://developer.apple.com/library/archive/qa/qa1801/_index.html

I recommend declaring the absence of subtitle content in the playlist by specifying CLOSED-CAPTIONS=NONE in the EXT-X-STREAM-INF tag.
